lightning/src/routing/network_graph.rs

```diff
@@ -1081,13 +1083,14 @@ impl NetworkGraph {
 	///
 	/// While there is no formal requirement that nodes regularly re-broadcast their channel
 	/// updates every two weeks, the non-normative section of BOLT 7 currently suggests that
-	/// pruning occurs for updates which are at least two weeks old, which we implement here.
+	/// pruning occur for updates which are at least two weeks old, which we implement here.
 	///
 	/// This function takes the current unix time as an argument. For users with the `std` feature
 	/// enabled, [`NetworkGraph::remove_stale_channels`] may be preferable.
 	pub fn remove_stale_channels_with_time(&self, current_time_unix: u64) {
 		let mut channels = self.channels.write().unwrap();
 		// Time out if we haven't received an update in at least 14 days.
+		if current_time_unix > u32::max_value() { return; } // Remove by 2106
 		let min_time_unix: u32 = (current_time_unix - STALE_CHANNEL_UPDATE_AGE_LIMIT_SECS) as u32;
```

Comment:
Seems since `current_time_unix` is user-supplied, it could be less than `STALE_CHANNEL_UPDATE_AGE_LIMIT_SECS` and underflow?